### PR TITLE
SmtpClient refactoring

### PR DESCRIPTION
**Pull request by apahim**
This is yet another refactoring to avoid the global initialization of
objects.

Signed-off-by: Amador Pahim <apahim@redhat.com>

### CODE REVIEW

**Comment by apahim** [comment]
Related to https://github.com/app-sre/qontract-reconcile/pull/1188

File: utils/smtp_client.py [hunk 1]
```diff
@@ -11,137 +11,81 @@
...
+
+        self.client = smtplib.SMTP(host=self.host,
+                                   port=self.port)
+        self.client.send
+        self.client.starttls()
+        self.client.login(self.user, self.passwd)
+
+        self.server = None
+        if not client_only:
+            self.server = self._init_server()
```
Comment (apahim) [inline utils/smtp_client.py:35]: this should actually be a lazy init. no reason to have the client_only path.

Comment (apahim) [reply utils/smtp_client.py:35]: will update the PR with that

**Review by rporres — APPROVED** [review]
YES! YES! <3

File: utils/smtp_client.py (hunk 1 above)
Comment (rporres) [reply utils/smtp_client.py]: you updated the PR just before I was going to suggest it 😂  Great!

**Comment by rporres** [comment]
Should we keep the teardown code in a destructor?

**Comment by rporres** [comment]
nah, QUIT is apparently useless https://cr.yp.to/smtp/quit.html

**Pull request merged**